hw-6/middlewares/check-auth-token.ts

```diff
@@ -0,0 +1,25 @@
+import { NextFunction, Request, Response } from 'express';
+import * as jwt from 'jsonwebtoken';
+import { HTTP_ERROR, JWT_SECRET } from '../constants/http-errors.enum';
+
+export const checkAuthToken = (req: Request, res: Response, next: NextFunction): void => {
+  const token = req.headers['x-access-token'] as string | undefined;
+
+  if (token) {
+    jwt.verify(token, JWT_SECRET, (err) => {
+      if (err) {
+        res.status(HTTP_ERROR.UNAUTHORIZED).json({
+          success: false,
+          message: 'Failed to authenticate token.',
+        });
+      } else {
+        next();
+      }
+    });
+  } else {
+    res.status(HTTP_ERROR.UNAUTHORIZED)
+      .send({ success: false, message: 'No token provided.' });
+  }
```

Comment:
We could separate the token validation onto a dedicated service or exported function in another module.

In this service/function it could return a promise instead of a callback.

This would allow to easily unit test it.

Reply:
Good point! I was thinking about it but decided it would be overkill for such a simple function. But I totally agree and in a real project, I'd do it the way you described. Thanks again!